sailfishusermanager.cpp

```diff
@@ -290,11 +291,20 @@ uint SailfishUserManager::addUser(const QString &name)
 
 int SailfishUserManager::removeUserFiles(uint uid)
 {
+    int rv = EXIT_FAILURE;
     QDir dir(USER_ENVIRONMENT_DIR.arg(uid));
     if (dir.removeRecursively())
-        return EXIT_SUCCESS;
-    qCWarning(lcSUM) << "Removing user environment directory failed";
-    return EXIT_FAILURE;
+        rv = EXIT_SUCCESS;
+    else
+        qCWarning(lcSUM) << "Removing user environment directory failed";
+
+    // Execute user removal scripts
+    QDir removal(USER_REMOVE_SCRIPT_DIR);
+    foreach (const QString &entry, removal.entryList(QDir::Files | QDir::NoDotAndDotDot)) {
+        QProcess::execute(USER_REMOVE_SCRIPT_DIR + '/' + entry, QStringList() << QString::number(uid));
```

Comment:
I think this should check if the process succeeded and if not, then print that to log. Maybe with command output if that's not already printed there (I didn't check!). It doesn't make sense to stop if a script fails but it should be clear and visible.

Reply:
Oh and also I think this could skip files without execution permissions. Could be handy if you can skip some of them that way. Just an idea, technically it probably works already as it shouldn't execute those anyway.